isis/src/base/apps/trackextract/trackextract.cpp

```diff
@@ -0,0 +1,213 @@
+  FileName cubeName = FileName(ui.GetFileName("TO"));
+  trackingName.setValue(cubeName.baseName() + "_tracking.cub"); //Strip off path and add _tracking
+  trackingGroup.addKeyword(trackingName);
+  mosaicLabel->addGroup(trackingGroup);
+
+  mosaicCube.close();
+}
+
+
+/**
+ * Creates the tracking cube by copying the input cube with only the tracking band.
+ * Then, goes through each pixel and subtracts the input cube's pixel type's min value
+ * because ProcesMosaic used to add the min value to each pixel.
+ *
+ * @param ui           The user interface
+ * @param trackingBand The index of the tracking band
+ */
+void createTrackCube(UserInterface ui, QString trackBand) {
+  ProcessByLine p;
+
+  CubeAttributeInput inAtt = CubeAttributeInput("+" + trackBand);
+  p.SetInputCube(ui.GetFileName("FROM"), inAtt);
+
+  FileName cubeName = FileName(ui.GetFileName("TO"));
+  // Strip off any extensions and add _tracking
+  QString trackingName = cubeName.path() + "/" + cubeName.baseName() + "_tracking.cub";
+  Cube inputCube = Cube(ui.GetFileName("FROM"));
+  int numSample = inputCube.sampleCount();
+  int numLine = inputCube.lineCount();
+
+  QString outAttString = "+UNSIGNEDINTEGER+" + QString::number(VALID_MINUI4) + ":";
+  outAttString += QString::number(VALID_MAXUI4);
+  CubeAttributeOutput outAtt = CubeAttributeOutput(outAttString);
+
+  p.SetOutputCube(trackingName, outAtt, numSample, numLine);
+
+  p.StartProcess(copyTrackPixels);
+  p.EndProcess();
+}
+
+
+/**
+ * Copies DN's from the input cube to the mosaic cube.
+ *
+ * @param in  Input cube
+ * @param out Mosaic cube
+ */
+void copyPixels(Buffer &in, Buffer &out) {
+  for (int i = 0; i < in.size(); i++) {
+    out[i] = in[i];
+  }
+}
+
+
+/**
+ * Copies DN's from the input cube to the tracking cube.
+ * Because each pixel is offsetted by the min value of the input cube's pixel type,
+ * we have to get that min value and subtract it from each pixel. Then, we add the min
+ * of an unsigned int to each pixel for the new offset.
+ * The default value is set in ProcessMosaic for pixels who are not from a cube.
+ * If a pixel's value is the default value, we set it to Null.
+ *
+ * @param in  Input cube
+ * @param out Tracking cube
+ */
+void copyTrackPixels(Buffer &in, Buffer &out) {
+  int offset = 0;
+  int defaultVal = 0;
+  switch (SizeOf(in.PixelType())) {
+    case 1:
+      offset = VALID_MIN1;
+      defaultVal = NULL1;
+      break;
+
+    case 2:
+      offset = VALID_MIN2;
+      defaultVal = NULL2;
+      break;
+
+    case 4:
+      offset = FLOAT_MIN;
+      defaultVal = INULL4;
+      break;
+
+    default:
+      QString msg = "Invalid Pixel Type [" + QString(in.PixelType()) + "]";
+      throw IException(IException::Programmer, msg, _FILEINFO_);
+  }
+
+  for (int i = 0; i < in.size(); i++) {
+    if (in[i] == (float) defaultVal) {
+      out[i] = NULLUI4;  // Set to the unsigned 4 byte Null value
+    }
+    else {
+      out[i] = ((int) in[i]) - offset + VALID_MINUI4;
```

Comment:
Should this be a qround call or an int cast? I'm not sure.

Reply:
We looked at qview, and they cast the pixel value to an int instead of using qround.

Reply:
That's probably the correct choice then.